File: ios/ArchiveSample/ArchiveSample/Person.m
//
//  Person.m
//  ArchiveSample
//
//  Created by 藤田 泰介 on 11/09/02.
//  Copyright (c) 2012年 Revolution 9. All rights reserved.
//

#import "Person.h"


@implementation Person

- (id)initWithCoder:(NSCoder *)decoder
{
    self = [super init];
    if (self) {
        _name = [decoder decodeObjectForKey:@"name"];
        _address = [decoder decodeObjectForKey:@"address"];
    }
    return self;
}

- (void)encodeWithCoder:(NSCoder *)encoder
{
    [encoder encodeObject:_name forKey:@"name"];
    [encoder encodeObject:_address forKey:@"address"];
}

- (void)dealloc
{
    self.name = nil;
    self.address = nil;
}

-(NSString*)description{
    return [NSString stringWithFormat:@"<%@ :%p , name:%@, address:%@>", 
            NSStringFromClass([self class]), self, _name, _address];
}

@end

File: ios_swift/Archive2/Archive2/Person.swift
//
//  Person.swift
//  Archive2
//
//  Created by taisuke fujita on 2014/07/10.
//  Copyright (c) 2014年 taisuke fujita. All rights reserved.
//

import UIKit

class Person: NSObject, NSCoding {
    // 名前
    let name: String
    // 住所
    let address: Address
    // コンストラクタ
    init(name: String, address: Address) {
        self.name = name
        self.address = address
    }
    // 復元用コンストラクタ
    required init(coder aDecoder: NSCoder) {
        self.name = aDecoder.decodeObjectForKey("name") as String
        self.address = aDecoder.decodeObjectForKey("address") as Address
    }
    // エンコード
    func encodeWithCoder(aCoder: NSCoder) {
        aCoder.encodeObject(self.name, forKey:"name")
        aCoder.encodeObject(self.address, forKey:"address")
    }
}
